File: app/src/main/java/com/annapol04/munchkin/db/AppDb.java
package com.annapol04.munchkin.db;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;

import com.annapol04.munchkin.data.GameDetails;
import com.annapol04.munchkin.data.HighscoreEntry;

@Database(entities = {HighscoreEntry.class, GameDetails.class}, version = 1, exportSchema = false)
public abstract class AppDb extends RoomDatabase {
    public abstract HighscoreEntryDao highscoreEntryDao();
    public abstract GameDetailsDao gameDetailsDao();
}

File: app/src/main/java/com/annapol04/munchkin/db/AppDb.kt
package com.annapol04.munchkin.db

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import android.arch.persistence.room.TypeConverters

import com.annapol04.munchkin.engine.Event

@Database(entities = arrayOf(Event::class), version = 1, exportSchema = false)
@TypeConverters(ActionTypeConverter::class, DataTypeConverter::class, ScopeTypeConverter::class, EventDataTypeConverter::class)
abstract class AppDb : RoomDatabase() {
    abstract fun eventDao(): EventDao
}
